src/pages/all-rules.js

```diff
@@ -0,0 +1,271 @@
+import Filter, { FilterOptions } from '../components/filter';
+import React, { useEffect, useState } from 'react';
+import {
+  faSortAmountDown,
+  faSortAmountUp,
+} from '@fortawesome/free-solid-svg-icons';
+
+import Breadcrumb from '../components/breadcrumb/breadcrumb';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Link } from 'gatsby';
+import SideBar from '../components/side-bar/side-bar';
+import { formatDistanceToNow } from 'date-fns';
+import { graphql } from 'gatsby';
+import locale from 'date-fns/locale/en-AU';
+import queryString from 'query-string';
+
+const AllRules = ({ data }) => {
+  const [filter, setFilter] = useState();
+  const [history, setHistory] = useState(data.allHistoryJson.edges);
+  const [rules, setRules] = useState(data.allMarkdownRemark.nodes);
+  const [notFound, setNotFound] = useState(false);
+  const [filterTitle, setFilterTitle] = useState('Results');
+  const [filteredItems, setFilteredItems] = useState({ list: [], filter: {} });
+  const [isAscending, setIsAscending] = useState(true);
+
+  const qs = queryString.parse(location.search, { parseNumbers: true });
+  const qsSize = qs.size ? (qs.size > 100 ? 100 : qs.size) : 10;
+
+  useEffect(() => {
+    filterBySort(filter);
+  }, [filter, isAscending]);
+
+  const filterBySort = (_filter) => {
+    if (!_filter || _filter?.length === 0) {
+      return;
+    }
+
+    setFilteredItems({});
+    let count = 0;
+    const foundRules = rules.map((item) => {
+      //Todo(Jack): Depending on the speed - optimise this find
+      let findRule = history.find(
+        (r) => sanitizeName(r) === sanitizeName(item.fields.slug, true)
+      );
+
+      //Return if a rule isn't found in history.json or if its archived
+      if (!findRule || item.frontmatter.archivedreason || count >= qsSize)
+        return;
+
+      count++;
+      return { item: item, file: findRule };
+    });
```

Comment:
In general, I think it would be better to break down logic like this into functions so as you read through you have an idea of what it is doing. Do you agree?

Reply:
Agreed, its quite a big function. Breaking it down would make it a lot more readable.